File: archive-common/src/main/java/site/archive/common/exception/common/DuplicateFieldValueException.java
package site.archive.common.exception.common;

import site.archive.common.exception.BaseException;
import site.archive.common.exception.ExceptionCode;

public class DuplicateFieldValueException extends BaseException {

    private static final String DUPLICATED_FILED_MESSAGE = "'%s'은 중복된 %s 입니다. 다른 %s을 사용해주세요.";

    public DuplicateFieldValueException(String duplicatedField, String duplicatedValue) {
        super(DUPLICATED_FILED_MESSAGE.formatted(duplicatedValue, duplicatedField, duplicatedField),
              ExceptionCode.DUPLICATED_FIELD_VALUE);
    }

}

File: archive-common/src/main/kotlin/site/archive/common/exception/common/DuplicateFieldValueException.kt
package site.archive.common.exception.common

import site.archive.common.exception.BaseException
import site.archive.common.exception.ExceptionCode

class DuplicateFieldValueException(duplicatedField: String, duplicatedValue: String) :
    BaseException(
        "'$duplicatedValue'은 중복된 $duplicatedField 입니다. 다른 $duplicatedField 을 사용해주세요",
        ExceptionCode.DUPLICATED_FIELD_VALUE
    )